images/gh-gl-sync/SpackCIBridge.py

```diff
@@ -153,10 +148,45 @@ def list_github_prs(self):
 
             if push:
                 # Check the PRs-to-be-pushed to see if any of them should be considered "backlogged".
-                # We currently recognize two types of backlogged PRs:
-                # 1) Some required "prerequisite checks" have not yet completed successfully.
-                # 2) Draft PRs. Handled earlier in this function.
-                if self.prereq_checks:
+                # We currently recognize three types of backlogged PRs:
+                # 1) The PR is based on a version of the "main branch" that has not yet been tested
+                # 2) Some required "prerequisite checks" have not yet completed successfully.
+                # 3) Draft PRs. Handled earlier in this function.
+                if self.main_branch:
+                    subprocess.run(["git", "fetch", "github", f"refs/pull/{pull.number}/head:{pr_string}"], check=True)
+                    # Get the merge base between this PR and develop.
+                    try:
+                        merge_base_sha = subprocess.run(
+                            ["git", "merge-base", pr_string, "github/develop"],
+                            check=True, stdout=subprocess.PIPE).stdout.strip()
+                    except subprocess.CalledProcessError:
+                        print(f"'git merge-base {pr_string} github/develop' returned non-zero. Skipping")
+                        self.unmergeable_shas.append(pull.head.sha)
+                        continue
+
+                    # Check if our PR's merge base is an ancestor of the latest tested main branch commit.
+                    if subprocess.run(
+                            ["git", "merge-base", "--is-ancestor", merge_base_sha, self.latest_tested_main_commit]
+                            ).returncode == 0:
+                        print(f"{pr_string}'s merge base IS an ancestor of latest_tested_main "
+                              "{merge_base_sha} vs. {self.latest_tested_main_commit}")
+                        try:
+                            subprocess.run(["git", "checkout", pr_string], check=True)
+                            subprocess.run(
+                                ["git", "merge", "-m", "Merge develop", self.latest_tested_main_commit],
+                                check=True)
```

Comment:
I wonder it we need to add `--no-ff` here, to make sure we always get a real merge commit with two parents?  If it's at all possible we end up with a fast-forward merge, I don't think the check I've added [here](https://github.com/spack/spackbot/pull/70/files#diff-32369261a6ca2d1e63555fead2186d95cc1825c7fcf9eee26b0b00b47a95b3a4R84) will work as intended (i.e. the `parent_ids` will only contain a single commit sha, the one from `develop`).

Reply:
I think you're right that we want to do `--no-ff` here. Let's experiment with it to make sure there aren't any unintended side-effects.